src/CruxComponent.tsx

```diff
@@ -85,51 +86,56 @@ export class CruxComponentCreator {
         class ListClass extends React.Component<any, any> {
 
             componentDidMount() {
-                this.fetchModels()
+                this.fetchModels(this.props)
```

Comment:
I am not getting this. Why are props being passed when they should be directly available inside any member function ?

Reply:
Yeah. It can be Accessed. But fetchModel has been also used in ComponentWillReceiveProps. Where we will be getting new props. So It is passed as params
```
componentWillReceiveProps(nextProps: any) {
                console.log("New props: ", nextProps, " ", this.props.queryParams)
                if (!_.isEqual(this.props.queryParams, nextProps.queryParams)) {
                    console.log("Fetching again : ", nextProps, this.props.queryParams)
                    this.fetchModels(nextProps)
                }
            }
```
It is implement as same as in lookfit

Reply:
this.props should give the access right ? I mean inside fetchModels, checkAdditionalModels, fetchServerData

Reply:
@ankitcf `this.props` is accessible everywhere, but that will always be current props, if `componentWillReceiveProps` needs to trigger `fetchModel`, then it has to be done for updated props, not current props.
If we don't pass updated props as function param, it does not have access to updated props, & fetch happens on stale props.
One way is to pass updated props(`nextProps`) as function argument.
Other way is to fetch in componentDidUpdate, where this.props will itself be updated.

Reply:
Got it. Can we only pass the data that is required instead of sending the whole props object.

Reply:
I think its better to pass complete props object, access is not an issue anyways.
creates lot of function params otherwise, and need to change every function signature when any one param changes.